### PR TITLE
Add Custom Price Query Type

### PR DESCRIPTION
**Pull request by spuddyminer**
This PR is intended to start the process of formalizing a new query type for any non-crypto asset like a stock, derivative product, or legacy market commodity. My take was to make it basically the same as the SpotPrice query type with an added parameter "context" that describes the nature of the spot price data that is to be reported.

Inspiration for this query type comes from communications with LandX. LandX is a protocol that tokenizes real world agricultural commodities. They are using 4 highly custom data points: CORN: per bushel, RICE: per CWT, SOYBEAN: per bushel, and WHEAT: per metric ton. An example of a CustomPrice query type for LandX is included in the proposed dataSpec.

Thanks for reading and review!    🙏

### CODE REVIEW

**Comment by themandalore** [comment]
@heavychain, should we future proof this and add an IPFS field?  If we're going down the route of the registry, should we just make all queryID's the IPFS field?  I think that ruins any readability right....pros/cons?

**Comment by tkernell** [comment]
> @heavychain, should we future proof this and add an IPFS field? If we're going down the route of the registry, should we just make all queryID's the IPFS field? I think that ruins any readability right....pros/cons?

I imagine that this dataSpecs repo and the on-chain registry could both exist in parallel, and we wouldn't necessarily need to change anything about the dataSpecs docs themselves, nor the queryType definitions or queryIDs. For now this repo hosts the canonical dataSpecs docs, and then when we switch, the on-chain registry will become the official source of the dataSpecs. At that point this repo could just become a staging area for writing dataSpecs, and act as a backup host of these docs.

File: types/CustomPrice.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Query Description
+
+This query returns the price of a non-crypto asset like a stock, derivatives product, or legacy market commodity.
+
+## Query Parameters
+
+The `CustomPrice` query has three parameters. The identifier parameter provides context for the data. The asset and currencty parameters specify the pricing pair.
+
+1. **identifier**
```
Comment (tkernell) [inline types/CustomPrice.md:15]: Maybe we should include some more example identifiers, such as: ForexPrice, IndexPrice, PreciousMetalPrice

**Comment by tkernell** [comment]
Nice structure Spuddy! I like the "context" arg approach for making this customizable.

**Pull request merged**